src/Qir/Execution/CommandLineCompiler/Program.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Help;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Tools;
+
+namespace CommandLineCompiler
+{
+    class Program
+    {
+        private static async Task<int> Main(string[] args)
+        {
+            var buildCommand = CreateBuildCommand();
+
+            var root = new RootCommand() { buildCommand };
+            SetSubCommandAsDefault(root, buildCommand);
+            root.Description = "Command-line tool for processing QIR DLL files.";
+            root.TreatUnmatchedTokensAsErrors = true;
+
+            Console.OutputEncoding = Encoding.UTF8;
+            return await new CommandLineBuilder(root)
+                .UseDefaults()
+                .UseHelpBuilder(context => new QsHelpBuilder(context.Console))
+                .Build()
+                .InvokeAsync(args);
+        }
+
+        /// <summary>
+        /// Creates the Build command for the command line compiler, which is used to
+        /// compile the executables from a given QIR DLL.
+        /// </summary>
+        /// <returns>The Build command.</returns>
+        private static Command CreateBuildCommand()
+        {
+            var buildCommand = new Command("build", "(default) Build the executables from a QIR DLL.")
+            {
+                Handler = CommandHandler.Create((BuildOptions settings) =>
+                {
+                    if (settings.LibraryDirectories.Length < 1)
+                    {
+                        throw new ArgumentException("The '--libraryDirectories' option requires at least one argument.");
+                    }
+                    return QirTools.BuildFromQSharpDll(settings.QsharpDll, settings.LibraryDirectories[0], settings.IncludeDirectory, settings.ExecutablesDirectory);
+                })
+            };
+            buildCommand.TreatUnmatchedTokensAsErrors = true;
+
+            Option<FileInfo> qsharpDllOption = new Option<FileInfo>(
```

Comment:
It would be nice if it was an either-or option of either giving the path to the qsharp dll or the path to the bitcode directly (or maybe even the human readable format would be nice as a third option). I haven't worked extensively with System.CommandLine, but usually it is possible to define alternative set of arguments. This is e.g. done for the Q# command line compiler by distinguishing code mode, snippet mode or responsefile mode (see [here](https://github.com/microsoft/qsharp-compiler/blob/5a90bed8554044d99d89953b922ddb8b804c4628/src/QsCompiler/CommandLineTool/Options.cs#L155)).

Reply:
I realize right now only having the path to the bitcode wouldn't work since you are creating the entry point object using the Q# AST. So I guess this is more my thinking for future evolution.

Reply:
I am unsure what is the distinction being drawn between the qsharp dll and the bitcode. The QIR bitcode is a resource found inside the dll, right? Or have I been doing this wrong so far?

Reply:
QIR bitcode is a resource found inside the DLL. I think @bettinaheim's comment is an idea on how this tool can evolve in the future but in the meantime I think it's ok to just have a `--qsharpDll` option.